package/MDAnalysis/analysis/dielectric.py

```diff
@@ -0,0 +1,148 @@
+                            {3 \varepsilon_ 0 V k_B T}
+
+for a system simulated in tin foil boundary conditions.
+"""
+
+from __future__ import division
+
+import numpy as np
+from numpy.testing import assert_almost_equal
+
+from MDAnalysis.units import constants, convert
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.mdamath import make_whole
+from MDAnalysis.due import due, Doi
+
+due.cite(Doi("10.1080/00268978300102721"),
+description="Dielectric analysis",
+path="MDAnalysis.analysis.dielectric",
+cite_module=True)
+del Doi
+
+class DielectricConstant(AnalysisBase):
+    r"""Dielectric constant
+
+    Parameters
+    ----------
+    selection : AtomGroup
+          any atomgroup
+    temperature : float
+        Temperature (Kelvin) at which the system has been simulated [298.15]
+    make_whole : bool
+        Make molecules whole; If the input already contains whole molecules
+        this can be disabled to gain speedup; the default is ``True``
+    verbose : bool
+          Show detailed progress of the calculation if set to ``True``; the
+          default is ``False``
+
+    Attributes
+    ----------
+    results : dict
+          * M: Directional dependant dipole moment
+            :math:`\langle \boldsymbol M \rangle` in :math:`eÅ`.
+          * M2: Directional dependant squared dipole moment
+            :math:`\langle \boldsymbol M^2 \rangle` in :math:`(eÅ)^2`
+          * fluct: Directional dependant dipole moment fluctuation
+            :math:`\langle \boldsymbol M^2 \rangle - \langle \boldsymbol M \rangle^2`
+            in :math:`(eÅ)^2`
+          * eps: Directional dependant static dielectric constant
+          * eps_mean: Static dielectric constant
+
+    Example
+    -------
+    Create a DielectricConstant object by supplying a selection,
+    then use the :meth:`run` method::
+
+      diel = DielectricConstant(selection)
+      diel.run()
+
+    The static dielectric constant of the system will be returned
+    within the `results` variable of the object.
+
+
+    .. versionadded:: 0.20.0
+
+    """
+    def __init__(self, selection, temperature=300, make_whole=True, **kwargs):
+        super(DielectricConstant, self).__init__(selection.universe.trajectory,
+                                                 **kwargs)
+        self.selection = selection
+        self.temperature = temperature
+        self.charges = selection.charges
+        self.volume = 0
+
+        # Dictionary containing results
+        self.results = {"M":  np.zeros(3),
+                        "M2": np.zeros(3),
+                        "fluct": np.zeros(3),
+                        "eps": np.zeros(3),
+                        "eps_mean": 0}
+
+    def _prepare(self):
+        error = "ERROR: Total charge of the selection is not zero."
+        assert_almost_equal(0, np.sum(self.charges), err_msg=error)
+
+    def _single_frame(self):
+        # Make molecules whole
+        if make_whole:
```

Comment:
The test cannot the `False`. Here, `make_whole` refers to `MDAnalysis.lib.mdamath.make_whole` and a function always test as `True`. It should be `self.make_whole`, which should be defined in the `__init__`.

The fact that id did not make the tests fail indicate that it is not covered by the tests.

Reply:
I fixed this and added a test. But I didn't found a nice test file where this actually makes a different for the results.